frontend/src/reducer/user.js

```diff
@@ -46,11 +62,7 @@ export const defaultUser = {
 export default function user (state = defaultUser, action) {
   switch (action.type) {
     case `${SET}/${USER_CONNECTED}`:
-      return {
-        ...state,
-        ...serializeUser(action.user),
-        lang: serializeUser(action.user).lang || state.lang
-      }
+      return serializeUser({ ...unserializeUser(state), ...action.user })
```

Comment:
can't you write it like this:
``` jsx
return { ...state, ...serializeUser(action.user) }
```

Btw, for the SET, I think we should not keep the old state, we should discard it because it is used, for example, when logging off and logging in with a new user.

It's different from UPDATE user where we only change some properties of the same user and where we must keep the old state

Reply:
it would end up like this 
``` jsx
return { ...serializeUser(action.user) }
```

Reply:
See https://github.com/tracim/tracim/pull/3150#discussion_r437982602 @Skylsmoi